### DISCUSSION TITLE
Dummy plugs make Twinkle Tray not able to communicate with any monitor

### DISCUSSION THREAD

**Discussion by lucasadelino** [xanderfrangos/twinkle-tray#645] [Monitor Compatibility Issues]
Hello,

First of all, thanks so much for working on TwinkleTray. It is a wonderful piece of software.

I just came across a strange issue. I have used Twinkle Tray with zero issues for several weeks. Recently, I purchased a couple of dummy plugs ([these two](https://www.amazon.com/dp/B071CGCTMY)) to make it easier to stream games to my Steam Deck at different resolutions. 

The dummy plugs' brightness can't be adjusted in TwinkleTray, as expected (I'm pretty sure they lack DDC/CI support). But, if both dummy plugs are connected, TwinkleTray can't communicate over DDC/CI with _any_ other monitor. Crucially, this issue only happens when _both_ dummy plugs are connected. If only (any) one of them is inserted, TwinkleTray works as expected. What's even more puzzling is that if I boot my computer with only a single dummy plug connected, and only then insert the other dummy plug and refresh the displays on TwinkleTray, everything works: TT recognizes all monitors, and is able to change the brightness of the non-dummy monitors:

![tt1](https://github.com/xanderfrangos/twinkle-tray/assets/89277256/525cc8c0-f4ba-47f8-9cce-8fc511596d93)

 But if I reboot my computer, leaving the two dummy plugs inserted, TT is no longer able to read them once it boots back up.

Oddly, everything works normally using ClickMonitorDDC, which is what I'm using in the meantime (though I vastly prefer TT!).

For reference, my GPU is an AMD 6900xt.

Any idea what may be causing this?

**Comment by xanderfrangos** [comment]
It always seems to be AMD users with the most reported issues. I swear something is up with their drivers.

Anyway, do you always have the green check marks and "DDC/CI" as the "communication method", even when the dummy plugs cause Twinkle Tray to stop communicating? Or does the "communication method" change to "None" when it stops working? Also, if you right-click on the tray icon and hit "Refresh displays", does that change the behavior?

**Reply by lucasadelino** [reply]
Thanks for the quick reply! It changes to 'None' when it stops working, and refreshing displays doesn't fix it

**Reply by xanderfrangos** [reply]
@lucasadelino The beta version of v1.16.0 may fix this, or at least help me see what's going on. If you don't mind, try out this specific version of v1.16.0-beta4:
https://github.com/xanderfrangos/twinkle-tray/suites/15541414863/artifacts/886885111

Copy and paste the Raw Monitor Data for me so I can take a look:
![Screenshot 2023-08-28 130009](https://github.com/xanderfrangos/twinkle-tray/assets/33106561/c48cb843-42dd-475c-ba95-e3d72595d1d4)

You should make a backup of `settings.json` in `%appdata%\twinkle-tray`, in case you want to roll back to v1.15.4 (v1.16.0 makes one-way changes to the settings). This all assumes you're **not** using the Windows Store version. If you are, you can just quit the Windows Store version while you test the beta version (and uninstall it later).

**Reply by lucasadelino** [reply]
Just tried the beta version, and the issue still persists. I tried rebooting and rescanning after installing the new beta version, and also tried uninstalling TT and doing a fresh install of the beta, just to make sure. Here's the raw monitor data:

```
{
  "7&3658f4a&0&UID512": {
    "id": "\\\\?\\DISPLAY#TCT0270#7&3658f4a&0&UID512",
    "key": "7&3658f4a&0&UID512",
    "num": 0,
    "brightness": 50,
    "brightnessMax": 100,
    "brightnessRaw": 50,
    "type": "none",
    "connector": "unknown",
    "min": 0,
    "max": 100,
    "hwid": [
      "DISPLAY",
      "TCT0270",
      "7&3658f4a&0&UID512"
    ],
    "name": "DP1080P60",
    "serial": "0"
  },
  "7&3658f4a&0&UID516": {
    "id": "\\\\?\\DISPLAY#XMD009A#7&3658f4a&0&UID516",
    "key": "7&3658f4a&0&UID516",
    "num": 1,
    "brightness": 50,
    "brightnessMax": 100,
    "brightnessRaw": 50,
    "type": "none",
    "connector": "unknown",
    "min": 0,
    "max": 100,
    "hwid": [
      "DISPLAY",
      "XMD009A",
      "7&3658f4a&0&UID516"
    ],
    "name": "DPF90435",
    "serial": "0"
  },
  "7&3658f4a&0&UID520": {
    "id": "\\\\?\\\\\\?\\DISPLAY#DEL41D9#7&3658f4a&0&UID520",
    "key": "7&3658f4a&0&UID520",
    "num": 2,
    "brightness": 50,
    "brightnessMax": 100,
    "brightnessRaw": 50,
    "type": "none",
    "connector": "displayport_external",
    "min": 0,
    "max": 100,
    "hwid": [
      "\\\\?\\DISPLAY",
      "DEL41D9",
      "7&3658f4a&0&UID520",
      "{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}"
    ],
    "name": "DELL S2721DGF",
    "serial": "8BC19H3",
    "sourceID": 0,
    "scaling": "identity",
    "bounds": {
      "width": 2560,
      "height": 1440,
      "pixelFormat": 32,
      "position": {
        "x": 0,
        "y": 0
      }
    }
  },
  "7&3658f4a&0&UID524": {
    "id": "\\\\?\\\\\\?\\DISPLAY#DEL2011#7&3658f4a&0&UID524",
    "key": "7&3658f4a&0&UID524",
    "num": 3,
    "brightness": 50,
    "brightnessMax": 100,
    "brightnessRaw": 50,
    "type": "none",
    "connector": "hdmi",
    "min": 0,
    "max": 100,
    "hwid": [
      "\\\\?\\DISPLAY",
      "DEL2011",
      "7&3658f4a&0&UID524",
      "{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}"
    ],
    "name": "DELL D2421H",
    "serial": "JLY0K33",
    "sourceID": 1,
    "scaling": "identity",
    "bounds": {
      "width": 1920,
      "height": 1080,
      "pixelFormat": 32,
      "position": {
        "x": 2560,
        "y": -243
      }
    }
  }
}
```

**Reply by xanderfrangos** [reply]
@lucasadelino Hmm, I was hoping for a better result. Could you run the EXE here from Command Prompt and share the output?
https://github.com/hensm/node-ddcci/files/4487925/ConsoleApplication1.zip

**Reply by xanderfrangos** [reply]
Also, can you take a screenshot of the expanded list of monitors in ControlMyMonitor (https://www.nirsoft.net/utils/control_my_monitor.html)? Both this and the above EXE's info should help me identify what's going on.

![Screenshot 2023-08-29 144452](https://github.com/xanderfrangos/twinkle-tray/assets/33106561/30b0850e-82d4-48ff-be97-7637cb1799cb)

I have a feeling your issue is related to some other incompatibility issues that other users have reported with Twinkle Tray. So you may be helping several people if we can get this figured out.

**Reply by lucasadelino** [reply]
Here's the output of ConsoleApplication: https://pastebin.com/vMPMnRsx

And here's the list of monitors from ControlMyMonitor:
![monitorlist2](https://github.com/xanderfrangos/twinkle-tray/assets/89277256/43728169-4f33-4ac5-a7b5-1d87f2c8dd3f)

Here's another piece of the puzzle that I just realized: the bug I'm describing only happens when the two dummy plugs are _disconnected_ on Windows display settings. If they're both connected, TT works normally, even with the two dummy plugs inserted.

Here's the list of monitors from CMM with the two plugs connected:
![monitorlist](https://github.com/xanderfrangos/twinkle-tray/assets/89277256/421b7928-7100-4881-8ce4-eb64b4793e22)

Happy to send any other info as needed, just let me know

**Reply by xanderfrangos** [reply]
@lucasadelino That helps a lot, thanks! I think I see what the issue is, but I'm not 100% sure how to solve it. I've implemented the most obvious fix in this version: https://github.com/xanderfrangos/twinkle-tray/suites/15626177969/artifacts/891841903

Let me know if that solves it. 👍

**Reply by xanderfrangos** [reply]
Actually... that one may not fix it. This might work better. I'm sort of able to replicate the issue on my end.
https://github.com/xanderfrangos/twinkle-tray/suites/15658357577/artifacts/893809851

**Reply by xanderfrangos** [reply]
Okay, last one. 😅
https://github.com/xanderfrangos/twinkle-tray/suites/15671854912/artifacts/894544561

Startup will take a little longer than before. I rewrote how DDC/CI detection works to (hopefully) be more accurate.

**Reply by lucasadelino** [reply]
Sorry for the delay, but I'm happy to report that this version fixed the issue! I really appreciate your work on this, thanks so much!

**Reply by xanderfrangos** [reply]
Great to hear! You are likely not the only person this will help.

**Comment by lucasadelino** [comment]
Quick update: while this fixed my problem (in that TT now recognizes all monitors), I'm now also getting the same black screen bug as mentioned in issues #649, #567, and #642. I didn't have the issue before using the new version of TT.
